### PR TITLE
fix(input): prevent browsers from autofilling hidden inputs

### PR DESCRIPTION
**Pull request by jbedard**
@petebacondarwin

### CODE REVIEW

**Review by mgol — APPROVED** [review]

File: test/e2e/tests/input-hidden.spec.js [hunk 1]
```diff
@@ -14,4 +14,79 @@ describe('hidden thingy', function() {
...
+    expect(element(by.css('#input1')).getAttribute('value')).toEqual('{{ internalFn() }}');
+    expect(element(by.css('#input2')).getAttribute('value')).toEqual('{{ internalFn() }}');
+    expect(element(by.css('body')).getAttribute('class')).toBe('');
+
+    browser.refresh();
+    expect(element(by.css('#input1')).getAttribute('value')).toEqual('');
+    expect(element(by.css('#input2')).getAttribute('value')).toEqual('');
+  });
+
+  it('should prevent browser autofill on location.refresh', function() {
```
Comment (petebacondarwin) [inline test/e2e/tests/input-hidden.spec.js]: `location.reload`?

**Review by gkalpak — APPROVED** [review]

File: test/e2e/tests/input-hidden.spec.js [hunk 2]
```diff
@@ -14,4 +14,79 @@ describe('hidden thingy', function() {
...
+    element(by.css('textarea')).sendKeys('{{ internalFn() }}');
+
+    expect(element(by.css('#input1')).getAttribute('value')).toEqual('{{ internalFn() }}');
+    expect(element(by.css('#input2')).getAttribute('value')).toEqual('{{ internalFn() }}');
+    expect(element(by.css('body')).getAttribute('class')).toBe('');
+
+    browser.refresh();
+    expect(element(by.css('#input1')).getAttribute('value')).toEqual('');
+    expect(element(by.css('#input2')).getAttribute('value')).toEqual('');
+  });
```
[the remark this answers is not in the document]
Comment (petebacondarwin) [reply test/e2e/tests/input-hidden.spec.js]: Perhaps we should *only* check the class?

**Comment by jbedard** [comment]
With updated browsers: https://travis-ci.org/angular/angular.js/jobs/445834683 (Unrelated failures)
Without updated browsers (but with protractor+selenium updates): https://travis-ci.org/angular/angular.js/builds/446006176 (GREEN!)
Without the fix: https://travis-ci.org/angular/angular.js/builds/446009530 (FF fails)

We don't have the e2e tests running in IE atm?

**Comment by mgol** [comment]
> We don't have the e2e tests running in IE atm?

We don't. Protractor has many IE-related issues, enabling it now might not be that easy.

**Review by petebacondarwin — APPROVED** [review]

**Pull request merged**